### PR TITLE
Fixing Account's CopyFrom

### PR DESCRIPTION
**Pull request by StanleyGoldman**
Fix for #408.
Edited from #561 as per @grokys request.

### CODE REVIEW

File: src/GitHub.App/Models/Account.cs [hunk 1]
```diff
@@ -15,6 +15,8 @@ public class Account : ReactiveObject, IAccount
     {
         BitmapSource avatar;
 
+        IObservable<BitmapSource> BitmapSource;
+        private IDisposable _bitmapSourceSubscription;
```
Comment (shana) [inline src/GitHub.App/Models/Account.cs]: Private fields should be lowercase with no underscore prefix. Feel free to use `this` during constructor assignment if the constructor argument name matches the field it's being assigned to.

We also don't use the `private` keyword in the codebase (it is not by any means a consensual choice, but we're following it for consistency)

File: src/GitHub.App/Models/Account.cs [hunk 2]
```diff
@@ -30,9 +32,11 @@ public class Account : ReactiveObject, IAccount
             PrivateReposInPlan = privateRepositoryInPlanCount;
             IsOnFreePlan = privateRepositoryInPlanCount == 0;
             HasMaximumPrivateRepositories = OwnedPrivateRepos >= PrivateReposInPlan;
+            BitmapSource = bitmapSource;
 
-            bitmapSource.ObserveOn(RxApp.MainThreadScheduler)
-                .Subscribe(x => avatar = x);
+            _bitmapSourceSubscription = bitmapSource
+                .ObserveOn(RxApp.MainThreadScheduler)
+                .Subscribe(x => Avatar = x);
```
Comment (shana) [inline src/GitHub.App/Models/Account.cs]: The change that fixed the property setting is already on master so your branch is conflicting on it, need to resolve that before we can merge.

File: src/UnitTests/GitHub.App/Models/AccountModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+            Func<int, BitmapImage, IObservable<BitmapImage>> generateObservable = (seconds, image) =>
+            {
+                return Observable.Generate(
+                    initialState: 0, 
+                    condition: i => i < 1, 
+                    iterate: state => state + 1,
+                    resultSelector: i => image, 
+                    timeSelector: i => TimeSpan.FromSeconds(seconds));
+            };
```
Comment (shana) [inline src/UnitTests/GitHub.App/Models/AccountModelTests.cs]: Like @grokys mentioned in the other PR, timings are iffy, it's safer to use a `Subject` or perhaps a `ReplaySubject` and do `OnNext` on it whenever you want to push something. Saves on waits, too :)

**Comment by shana** [comment]
So I was wondering the consequences of disposing and resubscribing to the bitmap source so I went looking at the image cache that's serving this observable and oh boy oh boy is it a mess down there. Note to self, need to clean up the ImageCache code, it's very silly.

File: src/GitHub.App/Models/Account.cs (hunk 2 above)
Comment (StanleyGoldman) [inline src/GitHub.App/Models/Account.cs]: Rebased

File: src/GitHub.App/Models/Account.cs (hunk 1 above)
Comment (StanleyGoldman) [inline src/GitHub.App/Models/Account.cs]: Gotcha

File: src/UnitTests/GitHub.App/Models/AccountModelTests.cs (hunk 3 above)
Comment (StanleyGoldman) [inline src/UnitTests/GitHub.App/Models/AccountModelTests.cs]: Good call. I was able to write a better unit test without it.

File: src/UnitTests/GitHub.App/Models/AccountModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+            //Updating the account in the collection
+            col.AddItem(updatedAccount);
+
+            //Waiting for the collection to process the update
+            collectionEvent.WaitOne();
+            collectionEvent.Reset();
+
+            //Providing the second avatar
+            bitmapImageObservable.OnNext(orgImage);
```
Comment (StanleyGoldman) [inline src/UnitTests/GitHub.App/Models/AccountModelTests.cs]: Updated

**Comment by StanleyGoldman** [comment]
Rebased

**Comment by shana** [comment]
Woot! Thank you for this fix, apologies (again) for the delays, we're really shorthanded around here.

**Pull request merged**